prac_07/project_management.py

```diff
@@ -0,0 +1,183 @@
+
+
+def load_projects(filename):
+    """Load projects from file"""
+    projects = []
+    with open(filename, encoding="utf-8") as in_file:
+        in_file.readline()  # Remove header
+        for line in in_file:
+            parts = line.strip().split('\t')
+            start_date = datetime.datetime.strptime(parts[1], "%Y-%m-%d").date()
+            project = Project(parts[0], start_date, int(parts[2]), float(parts[3]), float(parts[4]))
+            projects.append(project)
+    return projects
+
+
+def save_projects(projects, filename):
+    """Save projects to tab delimited file"""
+    with open(filename, 'w', encoding="utf-8") as out_file:
+        print(HEADER, file=out_file)
+        for project in projects:
+            print(f"{project.name}\t{project.start_date}\t{project.priority}\t"
+                  f"{project.cost_estimate}\t{project.completion_estimate}", file=out_file)
+
+
+def display(projects):
+    """Display projects"""
+    print("Incomplete projects:")
+    incomplete_projects = [project for project in projects if not project.is_complete()]
+    incomplete_projects.sort()
+    for project in incomplete_projects:
+        print(f"\t{project}")
+    print("Complete projects:")
+    complete_projects = [project for project in projects if project.is_complete()]
+    complete_projects.sort()
+    for project in complete_projects:
+        print(f"\t{project}")
+
+
+def display_listed_projects(projects):  # add number to list
+    for project in enumerate(projects):
+        print(project[0], project[1])
+
+
+def add(projects):
+    """Add project"""
+    print("Let's add a new project")
+    name = input("Name: ")
+    start_date = get_valid_date("Start date (dd/mm/yyyy): ")
+    priority = get_valid_integer("Priority: ")
+    cost_estimate = get_valid_float("Cost estimate: ")
+    completion_estimate = get_valid_float("Percent complete: ")
+    project = Project(name, start_date, priority, cost_estimate, completion_estimate)
+    projects.append(project)
+
+
+def get_valid_date(prompt):
+    """Get valid date"""
+    valid_date = False
+    while not valid_date:
+        try:
+            start_date = input(prompt)
+            start_date = datetime.datetime.strptime(start_date, "%d/%m/%Y").date()
+            valid_date = True
+        except ValueError:
+            print("Invalid date")
+    return start_date
+
+
+def get_valid_integer(prompt):
+    """Get valid integer"""
+    valid_integer = False
+    while not valid_integer:
+        try:
+            integer = int(input(prompt))
+            valid_integer = True
+        except ValueError:
+            return ""
+    return integer
+
+
+def get_valid_float(prompt):
+    """Get valid float"""
+    valid_float = False
+    while not valid_float:
+        try:
+            float_value = float(input(prompt))
+            valid_float = True
+        except ValueError:
+            return ""
+    return float_value
+
```

Comment:
Your error checking for floats and integers returns an empty string if an incorrect value is inputted? In your add_project function, this means that a blank priority, cost estimate, and percent complete could be blank after adding a new project. Is this okay? 

It's only in the update project option where we want to be able to leave blank to retain existing values.